sympy/concrete/summations.py

```diff
@@ -1491,29 +1504,14 @@ def get_shift(denom):
         shift = - b / a / n
         return shift
 
-    #Need a dummy symbol with no assumptions set for get_residue_factor
-    z = Dummy('z')
-
-    def get_residue_factor(numer, denom, alternating):
+    z = Dummy('z')  # needed for residue calculation
+    def residues(numer, denom, alternating, roots):
         residue_factor = (numer.as_expr() / denom.as_expr()).subs(i, z)
         if not alternating:
             residue_factor *= cot(S.Pi * z)
         else:
             residue_factor *= csc(S.Pi * z)
-        return residue_factor
-
-    # We don't know how to deal with symbolic constants in summand
-    if f.free_symbols - {i}:
-        return None
-
-    if not (a.is_Integer or a in (S.Infinity, S.NegativeInfinity)):
-        return None
-    if not (b.is_Integer or b in (S.Infinity, S.NegativeInfinity)):
-        return None
-
-    # Quick exit heuristic for the sums which doesn't have infinite range
-    if a != S.NegativeInfinity and b != S.Infinity:
-        return None
+        return [residue(residue_factor.factor(extension=True), z, root) for root in roots]
```

Comment:
It should be extension=root rather than extension=True. I showed the wrong thing before.

Reply:
Feel free to modify residue calculations. All tests are now passing locally.